big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -85,6 +122,47 @@ cert_with_cn_no_xmpp_addrs_request_name_empty(C) ->
 
     escalus_connection:stop(Client).
 
+self_signed_cert_fails_to_authenticate_with_tls(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_tcp).
+
+self_signed_cert_fails_to_authenticate_with_ws(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_ws).
+
+self_signed_cert_fails_to_authenticate_with_bosh(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_bosh).
+
+self_signed_cert_fails_to_authenticate(C, EscalusTransport) ->
+    F = fun() ->
+		UserSpec = generate_user(C, "alice-self-signed", EscalusTransport),
+		{ok, Client, _} = escalus_connection:start(UserSpec),
+		escalus_connection:stop(Client)
+	end,
+    %% We spawn the process trying to connect because otherwise the testcase may crash
+    %% due linked process crash (client's process are started with start_link)
```

Comment:
Might be more idiomatic with `process_flag(trap_exit, true)`?

Reply:
Like in the spawned process or the test case's process?

Reply:
If you do that in the test case's process there should be no need for a separate process + monitor, because we'll get linked process' exit signal as a message instead of crashing.

Reply:
Yes, I know that. I simply didn't want to mess with common tests framework. There may be other process linked to the test case's process and if they crash the test case should crash as well. I decided to spawn new process and monitor it to eliminate possible issues caused by the `trap_exit` set to true for the test case process.

Reply:
I think we have a similar problem here, since `{'DOWN', MRef, process, Pid, _Reason}` is treated as a test success, if another linked process fails the test case will pass anyway

Reply:
You're right that other signals may mess with the test case process

Reply:
> I think we have a similar problem here, since {'DOWN', MRef, process, Pid, _Reason} is treated as a test success, if another linked process fails the test case will pass anyway.

No, here is a patter matching, so we are expecting the `DOWN` message for given `MRef` (Monitor reference) and `Pid`. No other `DOWN` will match here.